src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -29,28 +30,50 @@ public interface MatlabBuild {
-        FilePath targetWorkspace = new FilePath(launcher.getChannel(),
-                workspace.getRemote() + "/" + MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME);
+        FilePath targetWorkspace; 
         ProcStarter matlabLauncher;
         if (launcher.isUnix()) {
-            final String runnerScriptName = uniqueName + "/run_matlab_command.sh";
+            targetWorkspace = new FilePath(launcher.getChannel(),
+                workspace.getRemote() + "/" + MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME);
+
+            // Determine whether we're on Mac on Linux
+            ByteArrayOutputStream kernelStream = new ByteArrayOutputStream();
+            launcher.launch()
+                .cmds("uname")
+                .masks(true)
+                .stdout(kernelStream)
+                .join();
+
+            String binaryName;
+            String runnerName;
+            if (kernelStream.toString("UTF-8").contains("Linux")) {
+                binaryName = "glnxa64/run-matlab-command";
+                runnerName = uniqueName + "/run-matlab-command";
+            } else {
+                binaryName = "maci64/run-matlab-command";
+                runnerName = uniqueName + "/run-matlab-command";
+            }
+
             matlabLauncher = launcher.launch().envs(envVars);
-            matlabLauncher.cmds(MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME + "/" + runnerScriptName, matlabCommand).stdout(listener);
+            matlabLauncher.cmds(MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME + "/" + runnerName, matlabCommand, startupOpts).stdout(listener);
 
             // Copy runner .sh for linux platform in workspace.
-            copyFileInWorkspace(MatlabBuilderConstants.SHELL_RUNNER_SCRIPT, runnerScriptName,
-                    targetWorkspace);
+            copyFileInWorkspace(binaryName, runnerName, targetWorkspace);
         } else {
-            final String runnerScriptName = uniqueName + "\\run_matlab_command.bat";
-            launcher = launcher.decorateByPrefix("cmd.exe", "/C");
+            targetWorkspace = new FilePath(launcher.getChannel(),
+                workspace.getRemote() + "\\" + MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME);
+
+            final String runnerName = uniqueName + "\\run-matlab-command.exe";
+            // launcher = launcher.decorateByPrefix("cmd.exe", "/C");
             matlabLauncher = launcher.launch().envs(envVars);
-            matlabLauncher.cmds(MatlabBuilderConstants.TEMP_MATLAB_FOLDER_NAME + "\\" + runnerScriptName, "\"" + matlabCommand + "\"")
+            matlabLauncher.cmds(targetWorkspace.toString() + "\\" + runnerName, "\"" + matlabCommand + "\"", startupOpts)
                     .stdout(listener);
+
             // Copy runner.bat for Windows platform in workspace.
-            copyFileInWorkspace(MatlabBuilderConstants.BAT_RUNNER_SCRIPT, runnerScriptName,
+            copyFileInWorkspace("win64\\run-matlab-command.exe", runnerName,

```

Comment:
Hmm, I think that this is a good suggestion. The part that makes me hesitate here is that we're already having to hardcode in the program name on lines 54, 57, and 69. I don't really like keeping that information in two separate places, since it means if the program name were to change it would be easy for a user to just change the constant name and miss the fact that there are also hardcoded instances. 

I don't have a strong preference either way, it just feels a little bit odd to me to have a property in another file define one instance and a hardcoded string define another. There's also a slight readability sacrifice when future contributors have to cross reference between the file they're working on and the constants file but that's not a real issue in the long run. 

Also I realize I left a line of code commented out, I'll update the pull request to remove that (and also probably condense lines 54 and 57 to be one line after the if block).

What are your thoughts @davidbuzinski?

Reply:
I don't have a strong opinion unless it's being used all over the code. I think it's basically just in 1 or 2 places right? I'm fine wither way